trunk/NDHMS/Routing/module_channel_routing.F

```diff
@@ -305,19 +339,44 @@ subroutine SUBMUSKINGCUNGE(    &
             if(D .eq. 0.0) then
               print *, "FATAL ERROR: D is 0 in MUSKINGCUNGE", Km, X, dt,D
               call hydro_stop("In MUSKINGCUNGE() - D is 0.")
-           endif
+           endif 
+ 
+! Another block of old code 
+!           C1 =  (Km*X + dt/2)/D
+!           C2 =  (dt/2 - Km*X)/D
+!           C3 =  (Km*(1-X)-dt/2)/D
+!           C4 =  (ql*dt)/D                       !-- ql already multipled by the dx length
+!           C4 =  (ql*dt)/D - (ChannK * WPk  * dx)  !-- DY & LKR lat inflow + channel loss
+           !TML-- channel loss now uses modified WP (from KINEROS2)           
+
+!           if(R .le. 0.0) then 
+!              Qj_0 =  ((C1*qup)+(C2*quc)+(C3*qdp) + C4)
+!           else 
+!              if( (C4 .lt. 0.0) .and. (abs(C4) .gt. (C1*qup)+(C2*quc)+(C3*qdp)) )  then
+!               C4 = -((C1*qup)+(C2*quc)+(C3*qdp))
+!              endif
+!             Qj_0 =  ((C1*qup)+(C2*quc)+(C3*qdp) + C4) - ((1/n) * AREA * (R**(2./3.)) * sqrt(So)) !f(x)
+!           endif
+! End of old block, the if statement preventing C4 from exceeding (C1*qup)+(C2*quc)+(C3*qdp)) needs to 
+! be added below. I attempted to insert it below.
 
            C1 =  (Km*X + dt/2.000000)/D
            C2 =  (dt/2.0000 - Km*X)/D
            C3 =  (Km*(1.00000000-X)-dt/2.000000)/D
 !          C1 =  max(0.0,min(1.0,1.0-C3))
-           C4 =  (ql*dt)/D
-!          C4 =  (ql*dt)/D - (ChannK * (WP + WPC) * dx)  !-- DY & LKR lat inflow + channel loss
+!           C4 =  (ql*dt)/D
+           C4 =  (ql*dt)/D - (ChannK * (WP + WPC) * dx)  !-- DY & LKR lat inflow + channel loss
 
            if((WP+WPC) .gt. 0.0) then  !avoid divide by zero
+! Another sanity check, prevent channel loss from exceeding flow
+             if( (C4 .lt. 0.0) .and. (abs(C4) .gt. (C1*qup)+(C2*quc)+(C3*qdp)) )  then
+              C4 = -((C1*qup)+(C2*quc)+(C3*qdp))
+             endif
```

Comment:
Is this logic statement (preventing negative flow) still correct, or are there other issues we need to consider now with the compound channel?